packages/picasso.js/src/core/chart-components/labels/strategies/bars.js

```diff
@@ -32,9 +32,29 @@ function toBackground(label) {
   };
 }
 
-export function isTextInRect(rect, textMetrics, opts) {
-  return opts.rotate ? !(rect.width < textMetrics.height || rect.height < textMetrics.width)
-    : !(rect.width < textMetrics.width || rect.height < textMetrics.height);
+function isFitWidth(rect, label, rotate, ignoreCheck) {
+  return ignoreCheck || (rotate ? rect.width >= label.height : rect.width >= label.width);
+}
+
+function isFitHeight(rect, label, rotate, ignoreCheck) {
+  return ignoreCheck || (rotate ? rect.height >= label.width : rect.height >= label.height);
+}
+
+function isTextFitRect(orientation, rect, label, fitsHorizontally, overflow) {
```

Comment:
So what does isTextFitRect mean?

Reply:
And how does it differ from isTextInRect?

Reply:
isTextFitRect  takes different types of parameter than isTextInRect. In core, they will use the same things (isFitWidth and isFitHeight).

Reply:
I do realize that naming is very hard, but `isTextFitRect, isFitWidth and isFitHeight` doesn't really mean anything. So I suggest you try and be a bit more descriptive and name them after their purpose in plain english.

Reply:
Updated!